src/inmanta/server/agentmanager.py

```diff
@@ -853,6 +853,13 @@ async def request_parameter(self, env_id: uuid.UUID, resource_id: ResourceIdStr)
             if res is None:
                 return 404, {"message": "The resource has no recent version."}
 
+            if res.status in UNDEPLOYABLE_NAMES:
+                LOGGER.debug(
+                    "Ignore fact request for %s, resource is in an undeployable state.",
+                    resource_id,
+                )
+                return 404, {"message": "The resource is in an undeployable state."}
```

Comment:
I don't think 404 is appropriate here. Perhaps 405? But I'm honestly not very familiar with HTTP error codes so someone else might know better.

Reply:
405 seems more appropriate :+1:

Reply:
I don't think 405 is appropriate,because that seems to imply it would work for e.g. a  a PUT on a different URL. 

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405

also 

```
The server must generate an [Allow](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Allow) header field in a 405 status code response. The field must contain a list of methods that the target resource currently supports.
```

503 Service Unavailable is most appropriate I think